File: src/leetcode/278. First Bad Version/first_bad_version_278.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        long long int start = 0;
        long long int end = n; 
        long long int index;       

        while(start <= end ) {
            long long int mid = (start + end) / 2;

            if(isBadVersion(mid)) {
                index = mid;
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }

        return index;
    }
};

File: src/leetcode/278. First Bad Version/first_bad_version_278.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        start = 0
        end = n
        index = -1

        while start <= end:
            mid = int((start + end) / 2)

            if isBadVersion(mid):
                index = mid
                end = mid - 1
            else:
                start = mid + 1
        return index
